### PR TITLE
[CPP] Improve handling of  __cxa_pure_virtual

### PR DESCRIPTION
**Pull request by tkreuzer**
## Purpose

This PR adds cpprt library for GCC, too. This provides __cxa pure virtual. Previously we added this manually to a few C++ projects, but this can cause a conflict on GCC 13, which generates symbols that conflict with __cxa_pure_virtual, if there is a pure virtual function in a vtable. So the definition needs to be in a different compilation unit. Instead adding it to a library seemed the better approach.
Fixes GCC 13 build.

### CODE REVIEW

File: sdk/lib/cpprt/__cxa_pure_virtual.cpp [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/*
+ * PROJECT:     GCC c++ support library
```
Comment (binarymaster) [inline sdk/lib/cpprt/__cxa_pure_virtual.cpp:2]:
```suggestion
 * PROJECT:     GCC C++ support library
```
There are differences between headers of the files in this directory though.

**Review by binarymaster — APPROVED** [review]

**Comment by SergeGautherie** [comment]
Are the 3 other occurrences fine as is? (Why?)
https://git.reactos.org/?p=reactos.git&a=search&h=HEAD&st=grep&s=__cxa_pure_virtual

File: sdk/lib/cpprt/CMakeLists.txt [hunk 2]
```diff
@@ -3,20 +3,30 @@ include_directories(
...
+else()
+    list(APPEND SOURCE
+        __cxa_pure_virtual.cpp)
 endif()
 
 add_library(cpprt ${SOURCE} ${cpprt_asm})
 set_target_cpp_properties(cpprt WITH_EXCEPTIONS)
 add_dependencies(cpprt xdk)
+
+# on GCC builds, we need to link against libsupc++
```
Comment (HBelusca) [inline sdk/lib/cpprt/CMakeLists.txt:29]:
```suggestion
# On GCC builds, we need to link against libsupc++
```

File: sdk/lib/cpprt/CMakeLists.txt [hunk 3]
```diff
@@ -3,20 +3,30 @@ include_directories(
...
-  add_asm_files(cpprt_asm arm/cpprt.s)
+# We need different things, depending on the compiler
+if(MSVC)
+    list(APPEND SOURCE
+        ehvec.cpp
+        new_nothrow.cpp
+        terminate.cpp
+        typeinfo.cpp)
+    if(ARCH STREQUAL "i386")
+      add_asm_files(cpprt_asm i386/cpprt.s)
```
Comment (HBelusca) [inline sdk/lib/cpprt/CMakeLists.txt:14]: (note we usually indent with multiples of 4 spaces)

**Review by HBelusca — APPROVED** [review]

**Comment by tkreuzer** [comment]
> Are the 3 other occurrences fine as is? (Why?) https://git.reactos.org/?p=reactos.git&a=search&h=HEAD&st=grep&s=__cxa_pure_virtual

Yes, because that is kernel stuff.

**Pull request merged**